caluma/caluma_form/structure.py

```diff
@@ -140,35 +140,40 @@ def fields(self):
         return self._fields
 
     @property
-    def sub_forms(self):
+    def sub_forms(self) -> List[Field]:
         if self._sub_forms is None:
-            self._sub_forms = {
-                field.question.slug: field
+            self._sub_forms = [
+                field
                 for field in self.children()
                 if field.question.type == Question.TYPE_FORM
-            }
+            ] + [
+                child
+                for field in self.children()
+                for child in field.children()
+                if field.question.type == Question.TYPE_TABLE
+            ]
         return self._sub_forms
 
-    def get_field(
-        self, question_slug: str, check_parent: bool = True
-    ) -> Optional[Field]:
+    def get_fields(self, question_slug: str, check_parent: bool = True) -> List[Field]:
```

Comment:
Can this ever return more than one field?

Reply:
Yes, in case the question is in a table, there will be more than one answer for the question and thus multiple fields.

Reply:
Can you point me to the place in the code where this happens? I've re-read `get_fields` a few times and don't see where this would return more than one field. Sorry if i'm missing something trivial.

Reply:
Huh, actually you are right, this is not finished, although calculated fields on table questions should work.
We shall address this follow-up, I think I have a matching test case already.